plugin/storage/kafka/marshaller.go

```diff
@@ -15,44 +15,32 @@
 package kafka
 
 import (
-	"encoding/json"
+	"bytes"
 
-	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/gogo/protobuf/jsonpb"
 
 	"github.com/jaegertracing/jaeger/model"
-	"github.com/jaegertracing/jaeger/model/converter/thrift/jaeger"
 )
 
 // Marshaller encodes a span into a byte array to be sent to Kafka
 type Marshaller interface {
 	Marshal(*model.Span) ([]byte, error)
 }
 
-type thriftMarshaller struct {
-	tProtocolFactory *thrift.TBinaryProtocolFactory
+type jsonMarshaller struct {
+	pbMarshaller jsonpb.Marshaler
```

Comment:
what are the considerations for caching the marshaller?  Is it more efficient to reuse the same one? Is it thread-safe?

Reply:
Yes, the main consideration was efficiency. The struct fields are all config related and are not written to when marshalling so it is thread safe.